js/script.js

```diff
@@ -202,58 +223,82 @@ function notCollided() {
+    var landX = canvas.width - 50;
+    var landY = 335;
+    if (!startTime) // Check if first frame
+        startTime = time || performance.now(); // Update startTime
 
-    // deltaTime should be in the range [0 ~ 1]
+    // delataTime = portion of duration time that has passed, keeps track of current time
     var deltaTime = (time - startTime) / duration;
     // currentPos = previous position + (difference * deltaTime)
-    var currentX = itemX + ((canvas.width - 50 - itemX) * deltaTime);
-    var currentY = itemY + ((335 - itemY) * deltaTime);
-
-    if (deltaTime >= 1) { // Animation has finished
-        itemX = canvas.width - 50; // reset x variable
-        itemY = 335; // reset y variable
-        startTime = null; // reset startTime
+    // Move tomato to next 
+    var currentX = itemX + ((landX - itemX) * deltaTime);
+    var currentY = itemY + ((landY - itemY) * deltaTime);
+    if (deltaTime >= 1) { // Animation (pot -> platform) has finished
+        // Update x-coordiantes to where item landed
+        itemX = landX; //  End of glass platform
+        itemY = landY; // End of glass platform
+        startTime = null; // Reset startTime
+        // Draw tomato
         if (itemType == "tomato")
             ctx.drawImage(tomato, itemX, itemY);
-        else if (itemType == "fireKirby") {} // draw fire Kirby sprite
+        else if (itemType == "fireKirby") {} // Draw fire Kirby sprite
+        // Item has landed
         itemLanded = true;
-    } else {
+    } else { // Animation (pot -> platform) has not finished yet
         if (itemType == "tomato")
             ctx.drawImage(tomato, currentX, currentY);
-        else if (itemType == "fireKirby") {} // draw fire Kirby sprite
+        else if (itemType == "fireKirby") {} // Draw fire Kirby sprite
         requestAnimationFrame(landItem); // Continue with animation
     }
 }
 
-// Updates mirror with new meme
+/**
+ * Returns None
+ * updates currentmeme to a random Kirby meme image
+ */
 function updateMirror() {
-    var newrandommeme = memearray[Math.floor(Math.random() * memearray.length)];
-    while (currentmeme == newrandommeme && newrandommeme == meme0) {
-        newrandommeme = memearray[Math.floor(Math.random() * memearray.length)];
+    // Produce random array index number for memearray
+    let arrayIndex = Math.floor(Math.random() * memearray.length);
+    // Set newrandommeme to memearray[arrayIndex]
+    var newrandommeme = memearray[arrayIndex];
+    // If newrandommeme is the same as previous meme or if arrayIndex == 0 then randomize arrayIndex again
+    while (currentmeme === newrandommeme || arrayIndex == 0) {
+        arrayIndex = Math.floor(Math.random() * memearray.length);
+        newrandommeme = memearray[arrayIndex];
     }
+    // Update  currentmeme
     currentmeme = newrandommeme;
 
 }
-
+/**
+ * Returns None
+ * Handles double click on chef Kirby by triggering tomato animation and moving it from pot -> platform
```

Comment:
Should this say that this function handles clicks and item animations?

Reply:
Yes I just wrote a more elaborated description to make it more clear.